File: ServiceApp/src/main/java/com/example/ServiceApp/dto/AttendanceDto.java
package com.example.ServiceApp.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
public class AttendanceDto {
    private Long id;
    private Long userId;
    private String userFirstname;
    private String userLastname;
    private String userEmail;
    private LocalDate date;
    private LocalDateTime checkInTime;
    private LocalDateTime checkOutTime;
    private Double workedHours;
    private Double overtimeHours;
    private String notes;
    private Boolean isManual;
}

File: service-interface/src/app/dtos/attendanceDto.ts
export interface AttendanceDto {
  id?: number;
  userId?: number;
  userFirstname?: string;
  userLastname?: string;
  userEmail?: string;
  date?: string;
  checkInTime?: string;
  checkOutTime?: string;
  workedHours?: number;
  overtimeHours?: number;
  notes?: string;
  isManual?: boolean;
}

export interface AttendanceStatusDto {
  isCheckedIn?: boolean;
  checkInTime?: string;
  attendanceId?: number;
}

export interface ManualAttendanceDto {
  userId?: number;
  date?: string;
  checkInTime?: string;
  checkOutTime?: string;
  notes?: string;
}

export interface WorkScheduleDto {
  id?: number;
  startTime?: string;
  endTime?: string;
  breakMinutes?: number;
  standardWorkHours?: number;
  workDays?: string;
  weekendBonusMultiplier?: number;
  isActive?: boolean;
}

export interface AttendanceReportDto {
  userId?: number;
  userFirstname?: string;
  userLastname?: string;
  userEmail?: string;
  startDate?: string;
  endDate?: string;
  totalWorkedHours?: number;
  totalOvertimeHours?: number;
  totalDaysWorked?: number;
  attendances?: AttendanceDto[];
}
